src/app/view/viewManager.js

```diff
@@ -172,6 +172,49 @@ export default class ViewManager {
         this.resetToDefaultTool();
     }
 
+    /**
+     * Sets the initial state of the name map
+     * @memberof ViewManager
+     * @returns {void}
+     */
+    setNameMap() {
+        const newMap = new Map();
+        for (let i = 0; i < this.currentDevice.layers.length; i++) {
```

Comment:
I guess I just don't know what the nameMap is and how we can automatically set in the device object's lifecycle

Reply:
nameMap simply records what the highest number applied to a given name is (e.g.; "Port": 6, "Channel": 15). nameMap is used by Device.generateNewName(), which sees if a string is a key in nameMap to get a number which enables the next name. setNameMap() is used to set the initial state of the nameMap when a JSON file is loaded in.